### PR TITLE
Fixes #37958 - Add "Sync Dependencies" option to Ansible collection repositories

### PR DESCRIPTION
**Pull request by Thorben-D**
[Redmine issue](https://projects.theforeman.org/issues/37958)

This PR adds a new option, "Sync dependencies", to Ansible collection repositories. 

When **enabled** (default), Pulp will sync the collections given in the requirements file and their dependencies.

When **disabled**, Pulp will only sync the collections, without their dependencies.

On the backend side, this is done by passing `sync_dependencies` to pulp_ansible.
I decided to keep track of where this setting is enabled at the root repository level, should we ever want to extend this functionality past Ansible collections.

### CODE REVIEW

**Comment by sjha4** [comment]
Code looks good. Would you be able to provide an example live repo to test this with? We'll want to add some automation around testing this.

The failing tests on this are VCR tests. Atix folks have previously recorded VCRs but if you face any issues, let me know and I can help with that..

**Comment by Thorben-D** [comment]
The VCRs should be good now, but I am struggling a bit with the test.
Really all that needs to be checked is the inclusion of the `sync_dependencies` parameter in the request to the Pulp API.
Whether the parameter does what it says is tested in pulp_ansible.
Do you have a pointer for me as to where such a test should go?

**Comment by sjha4** [comment]
There are some helper methods you could use here in the sync test: https://github.com/Katello/katello/blob/12b5a3d2cd8f554629762b4a8aafc358cbee2ffb/test/actions/pulp3/orchestration/ansible_collection_sync_test.rb#L31

I imagine something like:

```
pulp_remote = @repo.backend_service(@primary).get_remote
assert_equal pulp_remote.sync_dependencies, @repo.sync_dependencies
```

File: test/fixtures/vcr_cassettes/actions/pulp3/ansible_collection_sync/sync.yml [hunk 1]
```diff
@@ -43,7 +51,11 @@ http_interactions:
       Cross-Origin-Opener-Policy:
       - same-origin
       Correlation-Id:
+<<<<<<< HEAD
       - 860cc24bec6e4aa8ad51bb36e23a67de
+=======
```
Comment (sjha4) [inline test/fixtures/vcr_cassettes/actions/pulp3/ansible_collection_sync/sync.yml:56]: The VCRs seem to have merge conflicts of some sort..What you can do is restore these files, rebase and then re-record the sync test..

**Comment by sbernhard** [comment]
@sjha4 ready for review again :-)

**Comment by bastian-src** [comment]
Tested latest changes, LGTM!

**Review by sbernhard — APPROVED** [review]

**Pull request merged**